packages/cli/src/cmds/stats/stats.ts

```diff
@@ -0,0 +1,337 @@
+            // It's a return.  A return's parent_id matches the call's id
+            const functionKey =
+              'thread_' + event.threadId + '_id_' + event.parentId;
+            if (
+              functionKey in functionExecutionTimes[fileName].functions &&
+              functionExecutionTimes[fileName].functions[functionKey]
+                .callNoReturnFound
+            ) {
+              functionExecutionTimes[fileName].functions[
+                functionKey
+              ].callNoReturnFound = false;
+              if (event.elapsedInstrumentationTime) {
+                functionExecutionTimes[fileName].functions[
+                  functionKey
+                ].elapsedInstrumentationTimeTotal +=
+                  event.elapsedInstrumentationTime;
+              }
+              functionExecutionTimes[fileName].functions[
+                functionKey
+              ].numberOfCalls += 1;
+            } // else something went wrong: found return with no call
+          }
+        });
+      });
+      verbose(v);
+
+      // now that all functions were collected, index them by function
+      // name, not by <thread_id,parent_id>
+      let totalFunctionExecutionTimes = {};
+      for (const fileName in functionExecutionTimes) {
+        for (const functionKey in functionExecutionTimes[fileName].functions) {
+          const name =
+            functionExecutionTimes[fileName].functions[functionKey].name;
+          if (!(name in totalFunctionExecutionTimes)) {
+            totalFunctionExecutionTimes[name] = {
+              elapsedInstrumentationTimeTotal: 0,
+              numberOfCalls: 0,
+              path: functionExecutionTimes[fileName].functions[functionKey]
+                .path,
+            };
+          }
+          totalFunctionExecutionTimes[name].elapsedInstrumentationTimeTotal +=
+            functionExecutionTimes[fileName].functions[
+              functionKey
+            ].elapsedInstrumentationTimeTotal;
+          totalFunctionExecutionTimes[name].numberOfCalls += 1;
+        }
+      }
+
+      // convert hash to array
+      let flatFunctionExecutionTimes: FunctionExecutionTime[] = [];
+      for (const name in totalFunctionExecutionTimes) {
+        flatFunctionExecutionTimes.push({
+          name: name,
+          elapsedInstrumentationTimeTotal:
+            totalFunctionExecutionTimes[name].elapsedInstrumentationTimeTotal,
+          numberOfCalls: totalFunctionExecutionTimes[name].numberOfCalls,
+          path: totalFunctionExecutionTimes[name].path,
+        });
+      }
+
+      return flatFunctionExecutionTimes;
+    }
+
+    async function sortExecutionTimes(
+      functionExecutionTimes: FunctionExecutionTime[]
+    ): Promise<FunctionExecutionTime[]> {
+      // sort the array
+      return functionExecutionTimes.sort(
+        (a, b) =>
+          b.elapsedInstrumentationTimeTotal - a.elapsedInstrumentationTimeTotal
+      );
+    }
+
+    async function showStats(): Promise<
```

Comment:
Could you add a `try...catch` block in this function and then add a telemetry event named `stats:error` or something, so we can see when this command fails?

Reply:
I went ahead and did this.

@dividedmind I believe adding a `try...catch` here is a band-aid.  `runCommand` sends Telemetry for three types of errors; this here sends Telemetry for only one.  I believe the right solution is to address the root of the problem: to understand why `Yargs.exit` is in the source code and remove it if it shouldn't be there so tests can be written than call it.